tests/workbenches/notebook-controller/conftest.py

```diff
@@ -0,0 +1,282 @@
+def kueue_enabled_notebook_namespace(
+    request: FixtureRequest,
+    admin_client: DynamicClient,
+    unprivileged_client: DynamicClient,
+) -> Generator[Namespace, Any, Any]:
+    """Create a namespace with Kueue label enabled for notebook workloads"""
+    from utilities.infra import create_ns
+
+    namespace_name = request.param["name"]
+    add_kueue_label = request.param.get("add-kueue-label", True)
+
+    with create_ns(
+        admin_client=admin_client,
+        name=namespace_name,
+        unprivileged_client=unprivileged_client,
+        teardown=True,
+        add_kueue_label=add_kueue_label,
+        pytest_request=request,
+    ) as ns:
+        yield ns
+
+
+@pytest.fixture(scope="class")
+def patched_kueue_manager_config(
+    admin_client: DynamicClient,
+    request: FixtureRequest,
+) -> Generator[ConfigMap | None, Any, Any]:
+    """Conditionally patch the kueue-manager-config ConfigMap based on Kueue installation scenario
+
+    This fixture:
+    1. Detects the Kueue installation scenario
+    2. For RHOAI managed scenario: patches ConfigMap and restarts deployment
+    3. For Red Hat Kueue scenario: skips patching (no config/deployment present)
+    4. Yields the ConfigMap (or None if skipped) for tests
+    5. On teardown: restores original config if it was patched
+    """
+    # Detect Kueue installation scenario
+    scenario = detect_kueue_installation_scenario(client=admin_client)
+    LOGGER.info(f"Detected Kueue installation scenario: {scenario}")
+
+    # Check if we should patch config for this scenario
+    if not should_patch_kueue_config(scenario):
+        LOGGER.info(f"Skipping kueue-manager-config patching for scenario: {scenario}")
+        yield None
+        return
+
+    namespace = py_config["applications_namespace"]
+    config_map_name = "kueue-manager-config"
+
+    # Get the existing ConfigMap
+    try:
+        config_map = ConfigMap(
+            client=admin_client,
+            name=config_map_name,
+            namespace=namespace,
+            ensure_exists=True,
+        )
+    except Exception as e:
+        LOGGER.warning(f"Could not find kueue-manager-config ConfigMap: {e}")
+        LOGGER.info("This is expected for Red Hat Kueue operator scenario")
+        yield None
+        return
+
+    # Store original data and annotations for restoration
+    original_data = config_map.instance.data.copy() if config_map.instance.data else {}
+    original_annotations = (
+        config_map.instance.metadata.annotations.copy() if config_map.instance.metadata.annotations else {}
+    )
+
+    LOGGER.info("Storing original kueue-manager-config data for restoration")
+
+    # Get current config data
+    current_data = config_map.instance.data or {}
+    config_yaml = current_data.get("controller_manager_config.yaml", "{}")
+
+    # Parse the existing configuration
+    try:
+        config_dict = yaml.safe_load(config_yaml) or {}
+    except yaml.YAMLError:
+        config_dict = {}
```

Comment:
When can this happen? Should not we fail then?

Reply:
true, updated